airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/assets/metadata.py

```diff
@@ -92,13 +94,14 @@ def compute_catalog_overrides(merged_df):
     return catalogs
 
 
+# todo change this return value
 def merge_into_metadata_definitions(id_field, connector_type, oss_connector_df, cloud_connector_df) -> pd.Series:
     merged_connectors = pd.merge(
         oss_connector_df, cloud_connector_df, on=id_field, how="outer", suffixes=(OSS_SUFFIX, CLOUD_SUFFIX), indicator=True
     )
     sanitized_connectors = merged_connectors.where(pd.notnull(merged_connectors), None)
 
-    def build_metadata(merged_df):
+    def build_metadata(merged_df: dict) -> PartialMetadataDefinition:
```

Comment:
```suggestion
    def build_metadata(connector_catalog_entry: dict) -> PartialMetadataDefinition:
```

I found it hard to understand what was happening in this function until I realized you're manipulating a dict and not a dataframe will all the catalog entry.
Please add a docstring to explain what the format of the expected input and where it comes from.